api/src/main/java/org/openmrs/module/pihcore/merge/PihPatientMergeActions.java

```diff
@@ -123,10 +123,30 @@ private void moveQueueEntriesNotAssociatedWithVisits(Patient preferred, Patient
         }
     }
 
+    /**
+     * move bed patient assignments from non-preferred to preferred patient
+     * If the preferred patient already has an active bed assignment, then active bed assignments
+     * (there should be no more than one) on the non-preferred patient will be voided
+     * @param preferred
+     * @param nonPreferred
+     */
     private void moveBedPatientAssignments(Patient preferred, Patient nonPreferred) {
+        boolean preferredPatientHasBedAssigned = bedManagementService.getBedPatientAssignmentByPatient(preferred.getUuid(), false).size() > 0;
         List<BedPatientAssignment> bpaList = bedManagementService.getBedPatientAssignmentByPatient(nonPreferred.getUuid(), true);
         for (BedPatientAssignment bpa : bpaList) {
             bpa.setPatient(preferred);
+            // ensure there is at most one active bed assignment per patient
+            if(preferredPatientHasBedAssigned && bpa.getEndDatetime() == null) {
```

Comment:
This doesn't seem sufficient.  Don't we need to ensure that it also doesn't overlap with the preferred patient's bed assignment, even if it has already ended?  i.e. it can't have a endDateTime > the startDateTime of the preferred assignment, right?